src/test/java/com/aws/greengrass/authorization/WildcardTrieTest.java

```diff
@@ -16,282 +16,72 @@
 class WildcardTrieTest {
     @Test
     void testGlobWildcardMatching() {
+        // no wildcards
         WildcardTrie rt = new WildcardTrie();
-        rt.add("abc*xyz*");
-        assertTrue(rt.matchesMQTT("abc1234asdxyz456" ));
-        assertTrue(rt.matchesMQTT("abcdxyz"));
-        assertFalse(rt.matchesMQTT("abc123xyz456/89"));
-        assertFalse(rt.matchesMQTT(""));
-
-        assertTrue(rt.matchesStandard("abc1234asdxyz456" ));
-        assertTrue(rt.matchesStandard("abcdxyz"));
-        assertFalse(rt.matchesStandard("abc123xyz456/89"));
-        assertFalse(rt.matchesStandard(""));
-
-        rt.add("*/def");
-        assertTrue(rt.matchesMQTT("/def"));
-        assertTrue(rt.matchesMQTT("12345/def"));
-        assertTrue(rt.matchesMQTT(null));
-        assertFalse(rt.matchesMQTT("2/3/def"));
-
-        assertTrue(rt.matchesStandard("/def"));
-        assertTrue(rt.matchesStandard("12345/def"));
-        assertTrue(rt.matchesStandard(null));
-        assertFalse(rt.matchesStandard("2/3/def"));
-
-        // Only '*' should match all resources
+        rt.add("nowildcard");
+        rt.add("nowildcard2");
+        assertTrue(rt.matches("nowildcard"));
+        assertTrue(rt.matches("nowildcard"));
+        assertFalse(rt.matches("topic"));
+
+        // Test wildcards in middle
+        rt.add("abc*xy*z");
+        assertTrue(rt.matches("abc123xyabc!@#$%^&*()_+-=z" ));
+        assertTrue(rt.matches("abcxyz"));
+        assertTrue(rt.matches("abcxy7895z"));
+        assertTrue(rt.matches("abc123xy90zABCz"));
+        assertTrue(rt.matches("abc123xy90zABCxyABCz"));
+
+        assertFalse(rt.matches("ab789xyz123"));
+        assertFalse(rt.matches("abc789xy56z123"));
+        assertFalse(rt.matches("abc123xy90zABCzz0"));
+        assertFalse(rt.matches("abc123yx90z"));
+        assertFalse(rt.matches(""));
+
+        // Test multiple terminal points
+        rt.add("abc*xy*23");
+        assertTrue(rt.matches("abc789xy56z123"));
+        rt.add("abc*xy*");
+        assertTrue(rt.matches("abc123xy90zABCzz0"));
+        rt.add("abc*yx*z");
+        assertTrue(rt.matches("abc123yx90z"));
+
+        // Test Edge wildcards
         WildcardTrie rt1 = new WildcardTrie();
-        rt1.add("*wer");
-        assertFalse(rt1.matchesMQTT("9999/88"));
-        rt1.add("*");
-        assertTrue(rt1.matchesMQTT("9999/88"));
-        assertTrue(rt1.matchesStandard("9999/88"));
-
-        WildcardTrie rt2 = new WildcardTrie();
-        rt2.add("a/*/*/d");
-        assertTrue(rt2.matchesMQTT("a/12/34/d"));
-        assertFalse(rt2.matchesMQTT("a/2/3/4/d"));
-
-        assertTrue(rt2.matchesStandard("a/12/34/d"));
-        assertFalse(rt2.matchesStandard("a/2/3/4/d"));
-
-        rt2.add("**/abc");
-        assertTrue(rt2.matchesMQTT("78/abc"));
-        assertFalse(rt2.matchesMQTT("7/8/abc"));
+        rt1.add("*qwe*90*");
```

Comment:
glob matches include matching 0 characters. Do we have a test for that? 
Such as "a*" should match "ab" and "a"

Reply:
Yes